src/cpp/fastdds/publisher/DataWriterHistory.hpp

```diff
@@ -219,6 +219,17 @@ class DataWriterHistory : public fastrtps::rtps::WriterHistory
             std::unique_lock<fastrtps::RecursiveTimedMutex>& lock,
             const std::chrono::time_point<std::chrono::steady_clock>& max_blocking_time);
 
+    /**
+     * @brief Set unacknowledged sample removed functor.
+     *
+     * @param functor Functor to be set.
+     */
+    void unacknowledged_sample_removed_functor(
```

Comment:
I think we should inject the functor in the constructor of DataWriterHistory.
This way we force the user of DataWriterHistory (i.e. DataWriterImpl) to always set the functor.

Reply:
Done in c8cde8cad